massa-async-pool/src/message.rs

```diff
@@ -18,25 +18,24 @@ use nom::error::{context, ContextError, ParseError};
 use nom::multi::length_data;
 use nom::sequence::tuple;
 use nom::{IResult, Parser};
+use num::rational::Ratio;
 use serde::{Deserialize, Serialize};
 use std::ops::Bound::{Excluded, Included};
 
 /// Unique identifier of a message.
 /// Also has the property of ordering by priority (highest first) following the triplet:
-/// `(rev(max_gas*gas_price), emission_slot, emission_index)`
-pub type AsyncMessageId = (std::cmp::Reverse<Amount>, Slot, u64);
+/// `(rev(Ratio(msg.fee, max(msg.max_gas,1))), emission_slot, emission_index)`
+pub type AsyncMessageId = (std::cmp::Reverse<Ratio<u64>>, Slot, u64);
```

Comment:
Why did you choose this data format instead of saving the result of a `/` that could be represented in `Amount` as the "number of coins by gas" ?

Reply:
/ is lossy because of roundings. Ratios keep perfect precision